code/modules/power/supermatter/supermatter.dm

```diff
@@ -450,6 +505,12 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 		l.hallucination += power * config_hallucination_power * D
 		l.hallucination = CLAMP(0, 200, l.hallucination)
 
+	// Updates the displacement effect
+	var/current_status = get_status()
+	if(current_status != last_status)
+		last_status = current_status
+		update_icon(UPDATE_OVERLAYS)
```

Comment:
You'd want to take these lines into a proc together, then call a proc here only.

Reply:
done